src/wrapper/clap-juce-wrapper.cpp

```diff
@@ -1684,104 +1683,185 @@ class ClapJuceWrapper : public clap::helpers::Plugin<
+
+        ~EditorWrapperComponent() override
+        {
+            if (editor != nullptr)
+            {
+                juce::PopupMenu::dismissAllActiveMenus();
+                editor->processor.editorBeingDeleted(editor.get());
+            }
+        }
+
+        void createEditor(juce::AudioProcessor &plugin)
+        {
+            editor.reset(plugin.createEditorIfNeeded());
+
+            if (editor != nullptr)
+            {
 #if JUCE_VERSION >= 0x060008
-    std::unique_ptr<juce::AudioProcessorEditorHostContext> editorHostContext;
+                editorHostContext =
+                    std::make_unique<EditorHostContext>(host, clapWrapper.clapIDByParamPtr);
+                editor->setHostContext(editorHostContext.get());
+#endif
+#if !JUCE_MAC
+                editor->setScaleFactor(clapWrapper.guiScaleFactor);
 #endif
-    bool implementsGui() const noexcept override { return processor->hasEditor(); }
-    bool guiCanResize() const noexcept override
-    {
-        if (editor)
-            return editor->isResizable();
-        return true;
-    }
 
-    /*
-     * guiAdjustSize is called before guiSetSize and given the option to
-     * reset the size the host hands to the subsequent setSize. This is a
-     * relatively naive and unsatisfactory initial implementation.
-     */
-    bool guiAdjustSize(uint32_t *w, uint32_t *h) noexcept override
-    {
-        if (!editor)
-            return false;
+                addAndMakeVisible(editor.get());
+                editor->setTopLeftPosition(0, 0);
 
-        if (!editor->isResizable())
-            return false;
+                lastBounds = getSizeToContainChild();
 
-        auto cst = editor->getConstrainer();
+                {
+                    const juce::ScopedValueSetter<bool> resizingParentSetter{resizingParent, true};
+                    setBounds(lastBounds);
+                }
+            }
+            else
+            {
+                // if hasEditor() returns true then createEditorIfNeeded has to return a valid
+                // editor
+                jassertfalse;
+            }
+        }
 
-        if (!cst)
-            return true; // we have no constraints. Whaever is fine!
+        juce::Rectangle<int> getSizeToContainChild()
+        {
+            if (editor != nullptr)
+                return getLocalArea(editor.get(), editor->getLocalBounds());
 
-        auto minW = (uint32_t)cst->getMinimumWidth();
-        auto maxW = (uint32_t)cst->getMaximumWidth();
-        auto minH = (uint32_t)cst->getMinimumHeight();
-        auto maxH = (uint32_t)cst->getMaximumHeight();
+            return {};
+        }
 
-        // There is no std::clamp in c++14
-        auto width = juce::jlimit(minW, maxW, *w);
-        auto height = juce::jlimit(minH, maxH, *h);
+        static juce::Rectangle<int> convertToHostBounds(juce::Rectangle<int> pluginRect)
+        {
+            const auto desktopScale = juce::Desktop::getInstance().getGlobalScaleFactor();
```

Comment:
I copied this from the JUCE VST3 wrapper, but I wonder if we should prefer to use the scale factor requested by the CLAP host instead? (I imagine in many cases it would be the same)

Reply:
I think we should use the one provided by gui_set_scale yeah; whether it is different or not is less relevant than if it is what the host wants. An assert if they are different might be interesting though.